dashboard/src/actions/overviewActions.js

```diff
@@ -290,17 +305,23 @@ export const editMetadata =
 
     const rIndex = data.findIndex((item) => item.resource_id === rId);
     data[rIndex][metadata] = value;
-    data[rIndex][CONSTANTS.IS_DIRTY] = true;
-    if (value.length > CONSTANTS.DATASET_NAME_LENGTH) {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
-      data[rIndex][
-        CONSTANTS.NAME_ERROR_MSG
-      ] = `Length should be < ${CONSTANTS.DATASET_NAME_LENGTH}`;
-    } else if (value.length === 0) {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
-      data[rIndex][CONSTANTS.NAME_ERROR_MSG] = `Length cannot be 0`;
-    } else {
-      data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.SUCCESS;
+
+    if (metadata === CONSTANTS.NAME_KEY) {
+      if (value.length > CONSTANTS.DATASET_NAME_LENGTH) {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
+        data[rIndex][
+          CONSTANTS.NAME_ERROR_MSG
+        ] = `Length should be < ${CONSTANTS.DATASET_NAME_LENGTH}`;
+      } else if (value.length === 0) {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.ERROR;
+        data[rIndex][CONSTANTS.NAME_ERROR_MSG] = `Length cannot be 0`;
+      } else {
+        data[rIndex][CONSTANTS.NAME_VALIDATED] = CONSTANTS.SUCCESS;
+      }
+      data[rIndex][CONSTANTS.IS_DIRTY][CONSTANTS.DATASET_NAME_KEY] = true;
+    } else if (CONSTANTS.SERVER_DELETION_KEY) {
+      data[rIndex][CONSTANTS.IS_DIRTY][CONSTANTS.SERVER_DELETION_KEY] = true;
+      data[rIndex].metadata[CONSTANTS.SERVER_DELETION] = value;
```

Comment:
I'm confused about what we're doing at line 307 versus what we're doing at line 324.  In the case of `metadata === CONSTANTS.SERVER_DELETION` we're storing `value` twice in two different places:  in `data[i].serverDelete` and, I think, in `data[i].metadata.serverDelete`...and I think we're only supposed to do the first one (I don't think we're supposed to have the assignment at line 324).

Trying to maintain `name` and `serverDelete` at different levels in the data item results in awkward complexity in `getEditedMetadata()`.

Reply:
That's how they obtained from the response

Reply:
Line 306 does this:
```
    data[rIndex][metadata] = value;
```
then, if it's an expiration date, line 323 does this:
```
      data[rIndex].metadata[CONSTANTS.SERVER_DELETION] = value;
```
Why do we write the same value into two different places in the same object?

Reply:
Got it